R/unaids-navigator.R

```diff
@@ -53,16 +54,147 @@ write_navigator_checklist <- function(naomi_output,
+  }
+
+  ## Is scope set to national ID
+  national_area_id <- naomi_output$meta_area$area_id[naomi_output$meta_area$area_level == 0]
+  if (all(model_options$area_scope == national_area_id)) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_area_ID_selected"] <- TRUE
+  }
+
+  ## Check survey
+  # Does T1 calendar quarter match one survey
+  if (model_options$calendar_quarter_t1 %in% data_options$prev_survey_quarters) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_calendar_survey_match"] <- TRUE
+  }
+
+  ## Check that all surveys used are from most recent quarter available
+  most_recent_survey_available  <- max(data_options$prev_survey_available_quarters)
+  if (all(data_options$prev_survey_quarters == most_recent_survey_available)) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_recent_survey_only"] <- TRUE
+  }
+
+  # Is survey ART coverage included if available
+  if (length(data_options$artcov_survey_available) == 0 ||   ## Not available -> TRUE
+      data_options$artcov_survey_ids %in% data_options$artcov_survey_available) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_coverage"] <- TRUE
+  }
+
+  ## ## Check ART inputs
+  ## Is ART data included for at least one time point
+  has_art_t1 <- !is.null(naomi_output$fit$data_options$artnum_calendar_quarter_t1)
+  has_art_t2 <- !is.null(naomi_output$fit$data_options$artnum_calendar_quarter_t2)
+  
+  if ( has_art_t1 || has_art_t2 ) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_data"] <- TRUE
+  }
+
+  ## ## Check ANC inputs
+  ## Is ANC data included for at least one time point
+  if (!is.null(naomi_output$fit$data_options$anc_prev_year_t1)||
+      !is.null(naomi_output$fit$data_options$anc_prev_year_t2)) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ANC_data"] <- TRUE
+  }
+
+  ## ## Is ART attendance selected
+  ## * If ART data at both time points; artattend and artattend_t1 should be both TRUE
+  ## * If ART data at only one time point; only artattend should be TRUE
+  ## * If no ART data; both should be FALSE
+  if (has_art_t1 && has_art_t2) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- model_options[["artattend"]] && model_options[["artattend_t2"]]
+  } else if (has_art_t1 && !has_art_t2 || !has_art_t1 && has_art_t2 ) {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- model_options[["artattend"]] && !model_options[["artattend_t2"]]
+  } else {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Opt_ART_attendance_yes"] <- !model_options[["artattend"]] && !model_options[["artattend_t2"]]
+  }
+  
+
+  ## Check for correct calibration options selection
+  # Is logistic calibration selected
+
+  calibration_options <- naomi_output$fit$calibration_options
+
+  if (calibration_options$calibrate_method == "logistic") {
+    v$TrueFalse[v$NaomiCheckPermPrimKey == "Cal_method"] <- TRUE
+  }
+
+
+  if (all(naomi_output$meta_area$spectrum_region_code == 0)) {
```

Comment:
This is the kind of forking which we want a test to cover. Ideally every path should be run by the test code. So we want a test where `spec_level <- "nat"` and one where `spec_level <- "subnat"` so that both paths of the if on line 161 get run. Otherwise we are at risk of a silly typo in one of the names or something breaking the output

Reply:
Logic is fairly simple both sides so should be fine. Let's get this in